### PR TITLE
Say Sends Speech Locally Before Radio

### PR DESCRIPTION
**Pull request by Kurfursten**
Obsoletes #17138

The way this works is deceptively simple. Previously radio() would call talk_into directly. It still does, but I created a separate get_speech_flags() that does what radio() used to do but without the talk_into call. It's not really necessary that radio() still returns flags but it can't hurt. Also for the part with splitting up the list of listeners in send_speech, that was the best way to make sure it would handle radios after people, and it just werks.

Moderately tested but with saycode there are always some weird exceptions you wouldn't think of, probably. I did the most obvious stuff but I didn't test talking as a mouse, MMI, mecha, that sort of thing.

🆑 
* tweak: Speaking over a headset or radio will now cause your local speech to appear before the radio message.
* bugfix: Damage to telecomms will no longer cause your local messages to be delayed if you are standing in range of a radio. However, the radio transmission will still be delayed, and appear separately.

### CODE REVIEW

**Comment by DamianX** [comment]
The speech bubble will be delayed until the radio message arrives

**Comment by gbasood** [comment]
fix that

**Comment by DamianX** [comment]
More specifically, talking through an SBR/intercom while telecomms integrity is damaged will delay your bubble. Also might send the message twice.
image

Intercoms and SBRs should be completely independant, but they aren't apparently, and they weren't even before this change

**Comment by Kurfursten** [comment]
Yes, I'm sure you're right about that, I want to do some more testing first to see if I catch any more bugs before fixing the speech bubble thing.

As for sending the message twice/out of order, we can chalk that one up to "it's a feeeaturrreee" since they are damaged comms equipment.

File: code/modules/mob/living/say.dm [hunk 1]
```diff
@@ -252,7 +257,12 @@ var/list/department_radio_keys = list(
 
 	var/rendered = render_speech(speech)
 
-	for (var/atom/movable/listener in listeners)
+	var/list/listening_nonmobs = listeners.Copy()
```
Comment (DamianX) [inline code/modules/mob/living/say.dm:260]: This is unnecessary, the for loop below creates a copy of `listeners` internally and you can safely remove entries from it

Comment (Kurfursten) [reply code/modules/mob/living/say.dm:260]: There's a second for loop that just wants the non-mobs though. That loop needs to come *after* this loop so it's not internal anymore.

Comment (DamianX) [reply code/modules/mob/living/say.dm:260]: That's fine, the for loop iterates over the copy but the code inside the loop removes elements from the actual `listeners`

Comment (Kurfursten) [reply code/modules/mob/living/say.dm:260]: We still need listeners to have mobs included in it for when we send_speech_bubble though

Comment (DamianX) [reply code/modules/mob/living/say.dm:260]: Oh, I totally missed that, my bad.

**Comment by DamianX** [comment]
You know what, this is fine, the fact that SBR/intercom signals go through telecomms is a different issue.

**Pull request merged**